reflex-dom-core/src/Reflex/Dom/Builder/Class.hs

```diff
@@ -150,11 +150,41 @@ class (Monad m, Reflex t, DomSpace (DomBuilderSpace m), MonadAdjust t m) => DomB
                    => m ()
   notReady = lift notReady
 
+{- 2017-05-19 dridus: Commented out per Ryan Trinkle. Note that this does not support mount state and possibly has issues with unready child handling.
+
 class DomBuilder t m => MountableDomBuilder t m where
   type DomFragment m :: *
   buildDomFragment :: m a -> m (DomFragment m, a)
   mountDomFragment :: DomFragment m -> Event t (DomFragment m) -> m ()
 
+-}
+
+-- |'MonadMountStatus' represents an action that can be aware of whether the corresponding DOM built by the action is present within the document yet or not.
```

Comment:
Would it make sense to call it a 'widget' here, instead of an 'action'?  'MonadAdjust' does kind of imply that any sort of action may be adjusted, and that's deeply connected to this, but I'm not quite clear on what it'd mean outside the contexts of widgets anyhow.